warehouse/query-core/src/main/java/datawave/query/util/sortedset/BufferedFileBackedSortedSet.java

```diff
@@ -156,56 +159,74 @@ public boolean containsAll(Collection<?> c) {
             if (set.getSets().size() > 1) {
                 persist();
             }
-            compact(maxOpenFiles);
         } catch (IOException ioe) {
-            throw new RuntimeException("Unable to compact file backed sorted set", ioe);
+            throw new RuntimeException("Unable to persist or compact file backed sorted set", ioe);
         }
         return set.iterator();
     }
     
-    public void compact(int maxFiles) throws IOException {
+    /**
+     * If the number of sets is over maxFiles, then start compacting those files down. The goal is to get the number of files down around 50% of maxFiles in N
+     * rounds.
+     * 
+     * @param maxFiles
+     * @param rounds
+     * @throws IOException
+     */
+    public void compact(int maxFiles, int rounds) throws IOException {
         // if we have more sets than we are allowed, then we need to compact this down
         if (maxFiles > 0 && set.getSets().size() > maxFiles) {
             if (log.isDebugEnabled()) {
                 log.debug("Compacting " + handlerFactory);
             }
-            while (set.getSets().size() > maxFiles) {
-                List<SortedSet<E>> sets = set.getSets();
-                int numSets = sets.size();
-                // we want to do layers of compactions up to ivaratorMaxOpenFiles at a time until the total
-                // number of files is under maxFiles
-                int setsPerCompaction = Math.max(2, Math.min(maxOpenFiles, Math.round((float) numSets / maxFiles)));
-                int iterationsWithExtra = numSets / setsPerCompaction;
+            // create a copy of the set list (sorting below)
+            List<SortedSet<E>> sets = new ArrayList<>(set.getSets());
+            
+            // calculate the number of sets to compact per round
+            int numSets = sets.size();
+            int excessSets = numSets - (maxFiles / 2); // those over 50% of maxFiles
+            int filesToCompact = excessSets + rounds; // Add in a few extra to account for the compacted sets being added back in
```

Comment:
What is rounds is very large? Maybe we need some safety checks here.

Reply:
agreed... I will work that

Reply:
Also going to remove the concept of "rounds" since it is hard coded to 1.